lib/Doctrine/Common/Proxy/Autoloader.php

```diff
@@ -49,7 +49,11 @@ public static function resolveFile($proxyDir, $proxyNamespace, $className)
             throw InvalidArgumentException::notProxyClass($className, $proxyNamespace);
         }
 
-        $className = str_replace('\\', '', substr($className, strlen($proxyNamespace) + 1));
+        // remove proxy namespace from class name
+        $className = substr($className, strlen($proxyNamespace));
```

Comment:
Doesn't seem to make a difference. Humbug mutated this one to -1 i think and test were still green. I even added another dataset for the test to be sure.